### PR TITLE
grpc: add a warning for unsupported codec

### PR DESCRIPTION
**Pull request by easwars**
RELEASE NOTES:
- grpc: add a warning message for unsupported codecs; will stop defaulting to `proto` codec in a future release

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -1915,6 +1915,7 @@ func (s *Server) getCodec(contentSubtype string) baseCodec {
 	}
 	codec := encoding.GetCodec(contentSubtype)
 	if codec == nil {
+		logger.Warningf("Unsupported codec %q. Defaulting to %q for now. This will start to fail in future releases.", contentSubtype, proto.Name)
```
Comment (dfawley) [inline server.go:1918]: Is there a similar code path in `handler_server.go` that we also need to change?

**Comment by dfawley** [comment]
One more thing: let's add a test where the client uses an unregistered codec on the server if one doesn't already exist.

File: server.go (hunk 1 above)
Comment (easwars) [reply server.go:1918]: Both the handler_server and the regular grpc server read the content-subtype from the incoming request before creating the stream, and set the `contentSubType` field in the stream type. So, once we are here, the two paths have already converged.

**Comment by easwars** [comment]
> One more thing: let's add a test where the client uses an unregistered codec on the server if one doesn't already exist.

Will add one. The codec tests in our e2e tests are a little bit of a mess. I will clean them up in a separate PR and then will add the test here.

**Review by dfawley — APPROVED** [review]

**Pull request merged**